nautobot/extras/datasources/git.py

```diff
@@ -108,32 +150,50 @@ def pull_git_repository_and_refresh_data(repository_pk, request, job_result_pk):
         job_result.set_status(JobResultStatusChoices.STATUS_ERRORED)
 
     finally:
-        if job_result.status not in JobResultStatusChoices.TERMINAL_STATE_CHOICES:
-            if JobLogEntry.objects.filter(job_result__pk=job_result.pk, log_level=LogLevelChoices.LOG_FAILURE).exists():
-                job_result.set_status(JobResultStatusChoices.STATUS_FAILED)
-            else:
-                job_result.set_status(JobResultStatusChoices.STATUS_COMPLETED)
+        log_job_result_final_status(job_result, "synchronization")
+
+
+@nautobot_task
+def git_repository_diff_origin_and_local(repository_pk, request, job_result_pk, **kwargs):
+    """
+    Worker function to run a dry run on a Git repository.
+    """
+    job_result, repository_record = get_job_result_and_repository_record(
+        repository_pk,
+        job_result_pk,
+        logger=logger,
+    )
+    if not repository_record:
+        return
+
+    job_result.log(f'Running a Dry Run on Git repository "{repository_record.name}"...', logger=logger)
+    job_result.set_status(JobResultStatusChoices.STATUS_RUNNING)
+    job_result.save()
+    try:
+        if not os.path.exists(settings.GIT_ROOT):
+            os.makedirs(settings.GIT_ROOT)
+
+        git_repository_dry_run(repository_record, job_result=job_result, logger=logger)
+
+    except Exception as exc:
         job_result.log(
-            f"Repository synchronization completed in {job_result.duration}",
-            level_choice=LogLevelChoices.LOG_INFO,
-            logger=logger,
+            f"Error while running a dry run on {repository_record.name}: {exc}",
+            level_choice=LogLevelChoices.LOG_FAILURE,
         )
-        job_result.save()
+        job_result.set_status(JobResultStatusChoices.STATUS_ERRORED)
 
+    finally:
+        log_job_result_final_status(job_result, "dry run")
 
-def ensure_git_repository(repository_record, job_result=None, logger=None, head=None):
-    """Ensure that the given Git repo is present, up-to-date, and has the correct branch selected.
 
-    Note that this function may be called independently of the `pull_git_repository_and_refresh_data` job,
-    such as to ensure that different Nautobot instances and/or worker instances all have a local copy of the same HEAD.
-
-    Args:
-      repository_record (GitRepository)
-      job_result (JobResult): Optional JobResult to store results into.
-      logger (logging.Logger): Optional Logger to additionally log results to.
-      head (str): Optional Git commit hash to check out instead of pulling branch latest.
+def get_repo_from_url_to_path_and_from_branch(repository_record):
+    """Returns the from_url, to_path and from_branch of a Git Repo
+    :return (list): [
```

Comment:
This return value isn't actually a list, but a tuple! Please don't forget to update this to the `namedtuple` object that gets returned!